ui/src/datasets.tsx

```diff
@@ -151,6 +151,8 @@ export function Datasets() {
     )),
   });
 
+  const [selectAllChecked, setSelectAllChecked] = useState(true);
```

Comment:
The size of this list won't be large enough to cause any efficiency issues. However, it can just be a variable instead of a function since the result won't change until the component is re-rendered anyways.